<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=edge"/>
<title>Test results - Class com.stehno.ersatz.impl.DigestSpec</title>
<link href="../css/base-style.css" rel="stylesheet" type="text/css"/>
<link href="../css/style.css" rel="stylesheet" type="text/css"/>
<script src="../js/report.js" type="text/javascript"></script>
</head>
<body>
<div id="content">
<h1>Class com.stehno.ersatz.impl.DigestSpec</h1>
<div class="breadcrumbs">
<a href="../index.html">all</a> &gt; 
<a href="../packages/com.stehno.ersatz.impl.html">com.stehno.ersatz.impl</a> &gt; DigestSpec</div>
<div id="summary">
<table>
<tr>
<td>
<div class="summaryGroup">
<table>
<tr>
<td>
<div class="infoBox" id="tests">
<div class="counter">2</div>
<p>tests</p>
</div>
</td>
<td>
<div class="infoBox" id="failures">
<div class="counter">0</div>
<p>failures</p>
</div>
</td>
<td>
<div class="infoBox" id="ignored">
<div class="counter">0</div>
<p>ignored</p>
</div>
</td>
<td>
<div class="infoBox" id="duration">
<div class="counter">0.844s</div>
<p>duration</p>
</div>
</td>
</tr>
</table>
</div>
</td>
<td>
<div class="infoBox success" id="successRate">
<div class="percent">100%</div>
<p>successful</p>
</div>
</td>
</tr>
</table>
</div>
<div id="tabs">
<ul class="tabLinks">
<li>
<a href="#tab0">Tests</a>
</li>
<li>
<a href="#tab1">Standard output</a>
</li>
<li>
<a href="#tab2">Standard error</a>
</li>
</ul>
<div id="tab0" class="tab">
<h2>Tests</h2>
<table>
<thead>
<tr>
<th>Test</th>
<th>Duration</th>
<th>Result</th>
</tr>
</thead>
<tr>
<td class="success">DIGEST auth</td>
<td class="success">0.203s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">DIGEST no-auth</td>
<td class="success">0.641s</td>
<td class="success">passed</td>
</tr>
</table>
</div>
<div id="tab1" class="tab">
<h2>Standard output</h2>
<span class="code">
<pre>07:27:47.985 [Test worker] DEBUG io.undertow - starting undertow server io.undertow.Undertow@41b6e6d7
07:27:48.022 [XNIO-2 I/O-1] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-1', selector sun.nio.ch.EPollSelectorImpl@7dd6c6df
07:27:48.036 [XNIO-2 I/O-2] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-2', selector sun.nio.ch.EPollSelectorImpl@12a5f20a
07:27:48.052 [XNIO-2 I/O-3] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-3', selector sun.nio.ch.EPollSelectorImpl@55006484
07:27:48.054 [XNIO-2 I/O-4] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-4', selector sun.nio.ch.EPollSelectorImpl@707e4f9c
07:27:48.055 [XNIO-2 I/O-5] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-5', selector sun.nio.ch.EPollSelectorImpl@433a2164
07:27:48.070 [XNIO-2 I/O-6] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-6', selector sun.nio.ch.EPollSelectorImpl@6024bf63
07:27:48.075 [XNIO-2 I/O-7] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-7', selector sun.nio.ch.EPollSelectorImpl@68643921
07:27:48.088 [XNIO-2 I/O-8] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-8', selector sun.nio.ch.EPollSelectorImpl@16b70ed4
07:27:48.095 [Test worker] DEBUG io.undertow - Configuring listener with protocol HTTP for interface localhost and port 0
07:27:48.100 [XNIO-2 Accept] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 Accept', selector sun.nio.ch.EPollSelectorImpl@98dd94a
07:27:48.293 [XNIO-2 task-1] DEBUG io.undertow.request.security - Setting authentication required for exchange HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36755]} response {}}
07:27:48.294 [XNIO-2 task-1] DEBUG io.undertow.request.security - Attempting to authenticate /secrets, authentication required: true
07:27:48.300 [XNIO-2 task-1] DEBUG io.undertow.request.security - Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.DigestAuthenticationMechanism@793276b6 for /secrets
07:27:48.301 [XNIO-2 task-1] DEBUG io.undertow.request.security - Sending authentication challenge for HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36755]} response {}}
07:27:48.336 [XNIO-2 task-1] DEBUG io.undertow.request.security - Authentication result was CHALLENGE_SENT for /secrets
07:27:48.373 [XNIO-2 task-2] DEBUG io.undertow.request.security - Setting authentication required for exchange HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;ZsPTGnJHzPUNMTU0NDM2MjA2ODMxMdqh8Al/H+mqoNtOIAOlOn4=&quot;, uri=&quot;/secrets&quot;, response=&quot;b3dc19bb89b4be2e9bafbb6714525637&quot;, qop=auth, nc=00000001, cnonce=&quot;0215ff9c640a13a6&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36755]} response {}}
07:27:48.374 [XNIO-2 task-2] DEBUG io.undertow.request.security - Attempting to authenticate /secrets, authentication required: true
07:27:48.389 [XNIO-2 task-2] DEBUG io.undertow.request.security - Found digest header Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;ZsPTGnJHzPUNMTU0NDM2MjA2ODMxMdqh8Al/H+mqoNtOIAOlOn4=&quot;, uri=&quot;/secrets&quot;, response=&quot;b3dc19bb89b4be2e9bafbb6714525637&quot;, qop=auth, nc=00000001, cnonce=&quot;0215ff9c640a13a6&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot; in HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;ZsPTGnJHzPUNMTU0NDM2MjA2ODMxMdqh8Al/H+mqoNtOIAOlOn4=&quot;, uri=&quot;/secrets&quot;, response=&quot;b3dc19bb89b4be2e9bafbb6714525637&quot;, qop=auth, nc=00000001, cnonce=&quot;0215ff9c640a13a6&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36755]} response {}}
07:27:48.419 [XNIO-2 task-2] DEBUG io.undertow.request.security - Authentication failed with message UT000038: Authentication failed, requested user name 'admin' and mechanism DIGEST for HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;ZsPTGnJHzPUNMTU0NDM2MjA2ODMxMdqh8Al/H+mqoNtOIAOlOn4=&quot;, uri=&quot;/secrets&quot;, response=&quot;b3dc19bb89b4be2e9bafbb6714525637&quot;, qop=auth, nc=00000001, cnonce=&quot;0215ff9c640a13a6&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36755]} response {}}
07:27:48.426 [XNIO-2 task-2] DEBUG io.undertow.request.security - Authentication outcome was NOT_AUTHENTICATED with method io.undertow.security.impl.DigestAuthenticationMechanism@793276b6 for /secrets
07:27:48.426 [XNIO-2 task-2] DEBUG io.undertow.request.security - Sending authentication challenge for HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;ZsPTGnJHzPUNMTU0NDM2MjA2ODMxMdqh8Al/H+mqoNtOIAOlOn4=&quot;, uri=&quot;/secrets&quot;, response=&quot;b3dc19bb89b4be2e9bafbb6714525637&quot;, qop=auth, nc=00000001, cnonce=&quot;0215ff9c640a13a6&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36755]} response {}}
07:27:48.427 [XNIO-2 task-2] DEBUG io.undertow.request.security - Authentication result was CHALLENGE_SENT for /secrets
07:27:48.431 [Test worker] DEBUG io.undertow - stopping undertow server io.undertow.Undertow@41b6e6d7
07:27:48.458 [Test worker] DEBUG io.undertow - starting undertow server io.undertow.Undertow@6a01563d
07:27:48.462 [XNIO-3 I/O-1] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-1', selector sun.nio.ch.EPollSelectorImpl@b5f3c06
07:27:48.477 [XNIO-3 I/O-2] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-2', selector sun.nio.ch.EPollSelectorImpl@74a9c914
07:27:48.500 [XNIO-3 I/O-3] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-3', selector sun.nio.ch.EPollSelectorImpl@7a82faf4
07:27:48.522 [XNIO-3 I/O-4] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-4', selector sun.nio.ch.EPollSelectorImpl@7019fe99
07:27:48.527 [XNIO-3 I/O-5] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-5', selector sun.nio.ch.EPollSelectorImpl@3a505fbe
07:27:48.527 [XNIO-3 I/O-7] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-7', selector sun.nio.ch.EPollSelectorImpl@70316b0c
07:27:48.528 [XNIO-3 I/O-6] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-6', selector sun.nio.ch.EPollSelectorImpl@2745facb
07:27:48.533 [XNIO-3 I/O-8] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-8', selector sun.nio.ch.EPollSelectorImpl@13346f0e
07:27:48.552 [Test worker] DEBUG io.undertow - Configuring listener with protocol HTTP for interface localhost and port 0
07:27:48.552 [XNIO-3 Accept] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 Accept', selector sun.nio.ch.EPollSelectorImpl@1ab3ed06
07:27:48.578 [XNIO-3 task-1] DEBUG io.undertow.request.security - Setting authentication required for exchange HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:32903]} response {}}
07:27:48.579 [XNIO-3 task-1] DEBUG io.undertow.request.security - Attempting to authenticate /secrets, authentication required: true
07:27:48.579 [XNIO-3 task-1] DEBUG io.undertow.request.security - Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.DigestAuthenticationMechanism@3ea07fa3 for /secrets
07:27:48.579 [XNIO-3 task-1] DEBUG io.undertow.request.security - Sending authentication challenge for HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:32903]} response {}}
07:27:48.580 [XNIO-3 task-1] DEBUG io.undertow.request.security - Authentication result was CHALLENGE_SENT for /secrets
07:27:48.591 [XNIO-3 task-2] DEBUG io.undertow.request.security - Setting authentication required for exchange HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;mPwRVnb3PA0NMTU0NDM2MjA2ODU3OdmNHbgXdlLMpvgwjxi5UFM=&quot;, uri=&quot;/secrets&quot;, response=&quot;ff2af39536e775a9eddb209adaa469b3&quot;, qop=auth, nc=00000001, cnonce=&quot;a5cddf4542a9a34d&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:32903]} response {}}
07:27:48.591 [XNIO-3 task-2] DEBUG io.undertow.request.security - Attempting to authenticate /secrets, authentication required: true
07:27:48.592 [XNIO-3 task-2] DEBUG io.undertow.request.security - Found digest header Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;mPwRVnb3PA0NMTU0NDM2MjA2ODU3OdmNHbgXdlLMpvgwjxi5UFM=&quot;, uri=&quot;/secrets&quot;, response=&quot;ff2af39536e775a9eddb209adaa469b3&quot;, qop=auth, nc=00000001, cnonce=&quot;a5cddf4542a9a34d&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot; in HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;mPwRVnb3PA0NMTU0NDM2MjA2ODU3OdmNHbgXdlLMpvgwjxi5UFM=&quot;, uri=&quot;/secrets&quot;, response=&quot;ff2af39536e775a9eddb209adaa469b3&quot;, qop=auth, nc=00000001, cnonce=&quot;a5cddf4542a9a34d&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:32903]} response {}}
07:27:48.619 [XNIO-3 task-2] DEBUG io.undertow.request.security - Authenticated as admin, roles [TESTER]
07:27:48.619 [XNIO-3 task-2] DEBUG io.undertow.request.security - Authentication outcome was AUTHENTICATED with method io.undertow.security.impl.DigestAuthenticationMechanism@3ea07fa3 for /secrets
07:27:48.619 [XNIO-3 task-2] DEBUG io.undertow.request.security - Authentication result was AUTHENTICATED for /secrets
07:27:48.619 [XNIO-3 task-2] DEBUG com.stehno.ersatz.ErsatzServer - Request: { GET /secrets (query=[:], headers={Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;mPwRVnb3PA0NMTU0NDM2MjA2ODU3OdmNHbgXdlLMpvgwjxi5UFM=&quot;, uri=&quot;/secrets&quot;, response=&quot;ff2af39536e775a9eddb209adaa469b3&quot;, qop=auth, nc=00000001, cnonce=&quot;a5cddf4542a9a34d&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:32903]}, cookies=[:]): &lt;empty&gt; }
07:27:48.620 [XNIO-3 task-2] DEBUG com.stehno.ersatz.ErsatzServer - Response({Authentication-Info=[nextnonce=&quot;mPwRVnb3PA0NMTU0NDM2MjA2ODU3OdmNHbgXdlLMpvgwjxi5UFM=&quot;,qop=&quot;auth&quot;,rspauth=&quot;14aaffbd4fa1527b70f296115cdd3995&quot;,cnonce=&quot;a5cddf4542a9a34d&quot;,nc=00000001]}): Something secure
07:27:48.622 [Test worker] DEBUG io.undertow - stopping undertow server io.undertow.Undertow@6a01563d
07:27:48.644 [XNIO-3 task-2] ERROR io.undertow.request - UT005071: Undertow request failed HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;mPwRVnb3PA0NMTU0NDM2MjA2ODU3OdmNHbgXdlLMpvgwjxi5UFM=&quot;, uri=&quot;/secrets&quot;, response=&quot;ff2af39536e775a9eddb209adaa469b3&quot;, qop=auth, nc=00000001, cnonce=&quot;a5cddf4542a9a34d&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:32903]} response {Content-Encoding=[gzip], Connection=[keep-alive], Authentication-Info=[nextnonce=&quot;mPwRVnb3PA0NMTU0NDM2MjA2ODU3OdmNHbgXdlLMpvgwjxi5UFM=&quot;,qop=&quot;auth&quot;,rspauth=&quot;14aaffbd4fa1527b70f296115cdd3995&quot;,cnonce=&quot;a5cddf4542a9a34d&quot;,nc=00000001], Content-Length=[36], Date=[Sun, 09 Dec 2018 13:27:47 GMT]}}
java.lang.NullPointerException: null
	at io.undertow.conduits.DeflatingStreamSinkConduit.performFlushIfRequired(DeflatingStreamSinkConduit.java:449)
	at io.undertow.conduits.DeflatingStreamSinkConduit.flush(DeflatingStreamSinkConduit.java:371)
	at org.xnio.conduits.ConduitStreamSinkChannel.flush(ConduitStreamSinkChannel.java:162)
	at io.undertow.channels.DetachableStreamSinkChannel.flush(DetachableStreamSinkChannel.java:119)
	at org.xnio.channels.Channels.flushBlocking(Channels.java:63)
	at io.undertow.io.UndertowOutputStream.close(UndertowOutputStream.java:347)
	at io.undertow.io.BlockingSenderImpl.close(BlockingSenderImpl.java:219)
	at io.undertow.io.DefaultIoCallback.onComplete(DefaultIoCallback.java:54)
	at io.undertow.io.BlockingSenderImpl.invokeOnComplete(BlockingSenderImpl.java:276)
	at io.undertow.io.BlockingSenderImpl.send(BlockingSenderImpl.java:132)
	at io.undertow.io.BlockingSenderImpl.send(BlockingSenderImpl.java:166)
	at com.stehno.ersatz.ErsatzServer.send(ErsatzServer.groovy:595)
	at com.stehno.ersatz.ErsatzServer.access$0(ErsatzServer.groovy)
	at com.stehno.ersatz.ErsatzServer$1.handleRequest(ErsatzServer.groovy:444)
	at io.undertow.server.handlers.HttpTraceHandler.handleRequest(HttpTraceHandler.java:70)
	at io.undertow.security.handlers.AuthenticationCallHandler.handleRequest(AuthenticationCallHandler.java:52)
	at io.undertow.security.handlers.AuthenticationConstraintHandler.handleRequest(AuthenticationConstraintHandler.java:53)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.encoding.EncodingHandler.handleRequest(EncodingHandler.java:72)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:360)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
</pre>
</span>
</div>
<div id="tab2" class="tab">
<h2>Standard error</h2>
<span class="code">
<pre>Dec 09, 2018 7:27:48 AM okhttp3.internal.platform.Platform log
WARNING: previous digest authentication with same nonce failed, returning null
</pre>
</span>
</div>
</div>
<div id="footer">
<p>
<div>
<label class="hidden" id="label-for-line-wrapping-toggle" for="line-wrapping-toggle">Wrap lines
<input id="line-wrapping-toggle" type="checkbox" autocomplete="off"/>
</label>
</div>Generated by 
<a href="http://www.gradle.org">Gradle 4.10.2</a> at Dec 9, 2018 7:27:55 AM</p>
</div>
</div>
</body>
</html>
